src/UriHelper.php

```diff
@@ -179,26 +193,34 @@ public static function getSrcSetUrlString($url, $size, $custom = null)
      * This method will then generate the right rokka URLs to get what you want, see
      * `\Rokka\Client\Tests\UriHelperTest::provideGetSrcSetUrl` for some examples and the expected returns.
      *
-     * @param UriInterface $url    The original rokka render URL to be adjusted
-     * @param string       $size   The size of the image, eg '300w' or '2x'
-     * @param null|string  $custom Any rokka options you'd like to add, or are a dpi identifier like '2x'
+     * @param UriInterface $url           The original rokka render URL to be adjusted
+     * @param string       $size          The size of the image, eg '300w' or '2x'
+     * @param null|string  $custom        Any rokka options you'd like to add, or are a dpi identifier like '2x'
+     * @param bool         $setWidthInUrl If false, don't set the width as stack operation option, we provide it in $custom, usually as parameter
+     *
+     * @throws \RuntimeException
      *
      * @return UriInterface
      */
-    public static function getSrcSetUrl(UriInterface $url, $size, $custom = null)
+    public static function getSrcSetUrl(UriInterface $url, $size, $custom = null, $setWidthInUrl = true)
     {
         $identifier = substr($size, -1, 1);
         $size = substr($size, 0, -1);
         if ('x' === $identifier) {
             $uri = self::addOptionsToUri($url, 'options-dpr-'.$size);
         } elseif ('w' === $identifier) {
```

Comment:
$identifier can never be both, no need for an else, just use a separate if. Makes it a bit more readable.

Reply:
There's an important else in that if/elsif block. but maybe a `switch` would make it clearer

Reply:
a switch it is